consensus/core/src/leader_scoring.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{cmp::Ordering, fmt::Debug, sync::Arc};
+
+use consensus_config::AuthorityIndex;
+
+use crate::{commit::CommitRange, context::Context};
+
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub(crate) struct ReputationScores {
+    /// Score per authority. Vec index is the `AuthorityIndex`.
+    pub scores_per_authority: Vec<u64>,
+    // The range of commits these scores were calculated from.
+    pub commit_range: CommitRange,
+}
+
+#[allow(unused)]
+impl ReputationScores {
+    pub(crate) fn new(commit_range: CommitRange, scores_per_authority: Vec<u64>) -> Self {
+        Self {
+            scores_per_authority,
+            commit_range,
+        }
+    }
+
+    // Returns the authorities in score descending order.
+    pub fn authorities_by_score_desc(&self, context: Arc<Context>) -> Vec<(AuthorityIndex, u64)> {
```

Comment:
Could we inject the `context` to the `ReputationScores` struct instead of doing this here?